scripts/core/world/world-status.js

```diff
@@ -33,8 +34,49 @@ define(
                 // TODO: Generate a list of possible WorldStatus combination
                 // the goal is to create "biomes"
             }
+
+            this.serialize = function() {
+                if (!this.serialized) {
+                    var result = '';
+                    _.forEach(this.status, function(value, key) {
+                        result += `${key}:${value},`;
+                    });
+                    this.serialized = result;
+                }
+
+                return this.serialized;
+            }
         };
 
+        WorldStatus.getAllPossibleType = function() {
+            // TODO
```

Comment:
for here i asked help on stack overflow: http://stackoverflow.com/questions/43971480/generate-all-possible-objects-from-a-list-of-constraints/43972247#43972247